### PR TITLE
Updates generator to work with new annotations. And refactors generator for more decoupling

### PR DESCRIPTION
**Pull request by robert-northmind**
## Description
Solves issue https://github.com/bmw-tech/widget_driver/issues/74
This change updates generator to use new versions of widget_driver_annotation.
This new version of the annotations replaces old annotations with new names.

Now the generator knows how to parse these new annotations correctly.

This PR also:

- Adds unit tests
- Refactors generator code to be more testable and decoupled
- Updates README and exmaple/README to reflect new changes

## Type of Change

- [ ] 🚀 New feature (non-breaking change)
- [ ] 🛠️ Bug fix (non-breaking change)
- [x] ⚠️ Breaking change (feature or bug fix which breaks existing behaviors/APIs)
- [x] 🏗️ Code refactor
- [ ] ⚙️ Build configuration change
- [x] 📝 Documentation
- [ ] 🧹 Chore / Housekeeping

### CODE REVIEW

File: widget_driver/pubspec.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ dependencies:
   flutter:
     sdk: flutter
   meta: ^1.7.0
-  widget_driver_annotation: ^0.0.1
+  widget_driver_annotation: 0.0.2
```
Comment (robert-northmind) [inline widget_driver/pubspec.yaml:15]: We need to have this hard coded for now.. Since otherwise the widget driver is grabbing the new annotations.. which have breaking changes.. In my next PR I am fixing this

**Review by ebajrami — APPROVED** [review]

File: widget_driver_generator/lib/src/utils/element_utils.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:source_gen/source_gen.dart';
+import 'package:widget_driver_annotation/widget_driver_annotation.dart';
+
+import 'type_utils.dart';
+
+/// A helper class for getting data out of elements
+class ElementUtils {
+  /// Checks if the current element has an annotation which is valid.
+  /// If it has a valid annotation, then that annotation type is return, else it returns null.
```
Comment (JanMichaelPeter) [inline widget_driver_generator/lib/src/utils/element_utils.dart:10]:
```suggestion
  /// If it has a valid annotation, then that annotation type is returned, else it returns null.
```

**Review by JanMichaelPeter — APPROVED** [review]

File: widget_driver_generator/lib/src/utils/element_utils.dart (hunk 2 above)
Comment (JanMichaelPeter) [reply widget_driver_generator/lib/src/utils/element_utils.dart:10]: low priority tbh. If there is nothing else comming up, just leave it

File: widget_driver/example/ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -350,7 +350,7 @@
 		};
 		249021D4217E4FDB00AE95B9 /* Profile */ = {
 			isa = XCBuildConfiguration;
-			baseConfigurationReference = F5286DBEB732EE185778753C /* Pods-Runner.profile.xcconfig */;
+			baseConfigurationReference = 7AFA3C8E1D35360C0083082E /* Release.xcconfig */;
```
Comment (robert-northmind) [inline widget_driver/example/ios/Runner.xcodeproj/project.pbxproj:353]: I could not build locally before changing these settings. That why these changes are there

**Review by 4ndrey — APPROVED** [review]

**Review by JanMichaelPeter — APPROVED** [review]
